components/brave_shields/browser/ad_block_component_service_manager.cc

```diff
@@ -251,6 +262,35 @@ void AdBlockComponentServiceManager::EnableFilterList(const std::string& uuid,
   UpdateFilterListPrefs(uuid, enabled);
 }
 
+void AdBlockComponentServiceManager::CheckAdBlockComponentsUpdate() {
```

Comment:
For https://github.com/brave/brave-browser/issues/35216, the UI will want to know when this process completes (so that it can show a spinner and updated text, and also an indication if the process fails).

Could this method accept a callback that gets executed when all component updates are done? Or is there some other approach we want to take?

Reply:
Right now `OnDemandUpdate` has a [hardcoded no-op callback](https://github.com/brave/brave-core/blob/master/chromium_src/chrome/browser/component_updater/component_updater_utils.cc#L19). I suppose this could be changed, but let's see if @atuchin-m has any thoughts on the API we should use.

Reply:
I believe we could add on demand callback in `BraveOnDemandUpdate`. Also no problem to use chromium `OnDemandUpdater` directly (a chromium [example](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/components/components_handler.cc;l=190-192?q=OnDemandUpdate&start=11))

Also installing or updating a components has a several stages.
1. Updating the component (including downloading .crx)
2. Unpacking .crx and registering the new version 
3. Processing the content by a client code (=the adblock code here).
The callback is about 2 (maybe 1), but definitely not about 3.
Please be sure that you're waiting for the right things.